### PR TITLE
Bugfix/lab and patient intake form issues

### PR DESCRIPTION
**Pull request by Shreets**
-> add placeholder to date type input in Lab result page and covid history page
-> remove duplicate date of diagnosis input at the end of Covid history form.
->remove place holder from weight input.
->filter extra input field coming in from api in allergy form page

### CODE REVIEW

File: src/pages/LabIntakeInput.js [hunk 1]
```diff
@@ -227,18 +234,19 @@ const LabIntakeInput = ({ labState, setLabState, labError, setLabError }) => {
             </div>
           </div>
         </div>
-        <div>
+        <div className="date-input">
           <label>Lab specimen drawn on</label>
           <input
             name="specimenDrawnDate"
-            className="date-of-lab-test"
+            className="date-of-lab-test .date-placeholder"
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:241]:
```suggestion
            className="date-of-lab-test date-placeholder"
```

Comment (Shreets) [reply src/pages/LabIntakeInput.js]: resolved

File: src/pages/LabIntakeInput.js [hunk 2]
```diff
@@ -22,13 +23,22 @@ const LabIntakeInput = ({ labState, setLabState, labError, setLabError }) => {
...
   const handleLabInputChange = (e) => {
-    const { name: item, value, min, max} = e.target;
+    const { name: item, value, min, max} = e.target;    
 
     setLabError(state => ({ ...state, [item]: (min && +value < +min) || (max && +value > +max) }));
     setLabState({ ...labState, [item]: value });
-    console.log("@toDo add validation", setLabError);
+    console.log("@toDo add validation", setLabError);   
+    if(item === "specimenDrawnDate"){
+      value !== ""? setDateClassName("has-value"): setDateClassName("")
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:32]:
```suggestion
      setDateClassName(value ? "has-value" : '')
```

File: src/pages/LabIntakeInput.js [hunk 3]
```diff
@@ -22,13 +23,22 @@ const LabIntakeInput = ({ labState, setLabState, labError, setLabError }) => {
...
+    }
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:33]:
```suggestion
    if(item === "specimenDrawnDate"){
      setDateClassName(value ? "has-value" : "")
    } 
```

File: src/pages/CovidHistory.js [hunk 4]
```diff
@@ -49,6 +52,30 @@ const CovidHistory = ({
...
+      setDoseOneClass("has-value");
+    } else {
+      setDoseOneClass("");
+    }
+
+    if (intakeState["dateOfDose2Vaccination"] !== "") {
+      setDoseTwoClass("has-value");
+    } else {
+      setDoseTwoClass("");
+    }
```
Comment (nastikue) [inline src/pages/CovidHistory.js:72]:
```suggestion
      setDiagnosisClass(intakeState["dateCovidBefore"] !== "" ? "has-value" : "");
      setDoseOneClass(intakeState["dateOfDose1Vaccination"] !== "" ? "has-value" : "");
      setDoseTwoClass(intakeState["dateOfDose2Vaccination"] !== "" ? "has-value" : "");
```

File: src/pages/CovidHistory.js [hunk 5]
```diff
@@ -224,40 +251,52 @@ const CovidHistory = ({
...
+                      <input
+                        className={
+                          history.field === "dateOfDose1Vaccination"
+                            ? doseOneClass
+                            : doseTwoClass
+                        }
+                        type="date"
+                        id={indx}
+                        name={history.field}
+                        placeholder="mm/dd/yyyy"
```
Comment (nastikue) [inline src/pages/CovidHistory.js:276]:
```suggestion
                        placeholder=DATE_FORMAT.mmddyyyy
```

File: src/pages/LabIntakeInput.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import React, { useRef } from "react";
+import React, { useState, useEffect,useRef } from "react";
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:1]:
```suggestion
import React, { useState, useEffect, useRef } from "react";
```

File: src/pages/CovidHistory.js (hunk 4 above)
Comment (Shreets) [reply src/pages/CovidHistory.js]: resolved

File: src/pages/LabIntakeInput.js (hunk 3 above)
Comment (Shreets) [reply src/pages/LabIntakeInput.js]: resolved

File: src/pages/CovidHistory.js (hunk 5 above)
Comment (Shreets) [reply src/pages/CovidHistory.js]: resolved

File: src/pages/LabIntakeInput.js (hunk 6 above)
Comment (Shreets) [reply src/pages/LabIntakeInput.js]: resolved

**Review by nastikue — APPROVED** [review]

**Pull request merged**